lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -266,26 +267,44 @@ def create(self, trans, payload, **kwd):
                                              and will copy the outputs of the previously executed step.
         """
         ways_to_create = set([
+            'archive_source',
             'workflow_id',
             'installed_repository_file',
             'from_history_id',
             'shared_workflow_id',
             'workflow',
         ])
-        if len(ways_to_create.intersection(payload)) == 0:
+        if len(ways_to_create.intersection(payload)) != 1:
             message = "One parameter among - %s - must be specified" % ", ".join(ways_to_create)
             raise exceptions.RequestParameterMissingException(message)
 
-        if len(ways_to_create.intersection(payload)) > 1:
-            message = "Only one parameter among - %s - must be specified" % ", ".join(ways_to_create)
-            raise exceptions.RequestParameterInvalidException(message)
-
         if 'installed_repository_file' in payload:
-            workflow_controller = trans.webapp.controllers['workflow']
-            result = workflow_controller.import_workflow(trans=trans,
-                                                         cntrller='api',
-                                                         **payload)
-            return result
+            installed_repository_file = payload.get('installed_repository_file', '')
+            workflow_file = open(installed_repository_file, 'rb')
+            workflow_data = workflow_file.read()
+            workflow_file.close()
```

Comment:
Can you use the `with` statement instead of explicitly closing the file, please?